SwrveSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -1250,18 +1248,29 @@ protected void initResources() {
      */
     protected void initCampaigns() {
         campaigns = new ArrayList<SwrveBaseCampaign>();
+        campaignsState = new HashMap<Integer, SwrveCampaignState>();
 
         try {
             String campaignsFromCache = cachedLocalStorage.getSecureCacheEntryForUser(userId, CAMPAIGN_CATEGORY, getUniqueKey());
             if (!SwrveHelper.isNullOrEmpty(campaignsFromCache)) {
                 JSONObject campaignsJson = new JSONObject(campaignsFromCache);
-                // Load campaign settings
-                String campaignSettingsFromCache = cachedLocalStorage.getCacheEntryForUser(userId, CAMPAIGN_SETTINGS_CATEGORY);
-                JSONObject campaignSettingsJson = null;
-                if (!SwrveHelper.isNullOrEmpty(campaignSettingsFromCache)) {
-                    campaignSettingsJson = new JSONObject(campaignSettingsFromCache);
+                // Load campaigns state
+                String campaignsStateFromCache = cachedLocalStorage.getCacheEntryForUser(userId, CAMPAIGNS_STATE_CATEGORY);
+                if (!SwrveHelper.isNullOrEmpty(campaignsStateFromCache)) {
+                    JSONObject campaignsStateJson = new JSONObject(campaignsStateFromCache);
+                    Iterator<String> campaignIdIterator = campaignsStateJson.keys();
+                    while(campaignIdIterator.hasNext()) {
+                        String campaignIdStr = campaignIdIterator.next();
+                        try {
+                            int campaignId = Integer.parseInt(campaignIdStr);
+                            SwrveCampaignState campaignState = new SwrveCampaignState(campaignsStateJson.getJSONObject(campaignIdStr));
+                            campaignsState.put(campaignId, campaignState);
+                        } catch(Exception exp) {
+                            SwrveLogger.e(LOG_TAG, "Could not load state for campaign " + campaignIdStr, exp);
+                        }
+                    }
```

Comment:
Perhaps separate the above block into a  smaller method? Makes it easier to unit test. And we could potentially call this from the TestApp and display it there..